### PR TITLE
New Presence: IMDbFilmIzle

### PR DESCRIPTION
**Pull request by Serhann**
image
image
image
image

### CODE REVIEW

File: websites/I/IMDbFilmIzle/dist/metadata.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  "service": "IMDbFilmizle",
+  "version": "1.0.0",
+  "logo": "https://i.imgur.com/rDIoqZb.png",
+  "thumbnail": "https://i.imgur.com/RdOXsqx.png",
+  "color": "#FFDB4C",
+  "tags": [
+    "filmler",
+    "film izle",
+    "dizi",
+    "dizi izle"
```
Comment (ririxichan) [inline websites/I/IMDbFilmIzle/dist/metadata.json:29]: remove spaces and add ``-`` or delete these tags.

File: websites/I/IMDbFilmIzle/iframe.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    (document.querySelector("#vplayer video") as HTMLVideoElement) ||
+    (document.querySelector("video") as HTMLVideoElement);
+    if (video) {
+      iframe.send({
+        error: false,
+        currentTime: video.currentTime,
+        duration: video.duration,
+        paused: video.paused
+      });
+    } else iframe.send({ error: true })
```
Comment (ririxichan) [inline websites/I/IMDbFilmIzle/iframe.ts:14]:
```javascript
} else {
```

File: websites/I/IMDbFilmIzle/presence.ts [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+};
+
+/**
+* Get Timestamps
+* @param {Number} videoTime Current video time seconds
+* @param {Number} videoDuration Video duration seconds
+*/
+function getTimestamps(videoTime: number, videoDuration: number): Array<number> {
+  var startTime = Date.now();
+  var endTime = Math.floor(startTime / 1000) - videoTime + videoDuration;
```
Comment (ririxichan) [inline websites/I/IMDbFilmIzle/presence.ts:34]: var -> const and combine them

File: websites/I/IMDbFilmIzle/presence.ts [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+      presenceData.state = "Video Yöneticisi";
+    } else if ((new URLSearchParams(window.location.search)).get('i') == 'e') {
+      presenceData.details = "Bir sayfaya göz atıyor:";
+      presenceData.state = "Bilgileri Düzenle";
+    }
+  }
+
+  // Movies & Series
+  if (document.querySelector("#t1 > li > div.ssag > h1")?.textContent && video?.currentTime && isVideoData) {
+    let timestamps = getTimestamps(Math.floor(video.currentTime), Math.floor(video.duration));
```
Comment (ririxichan) [inline websites/I/IMDbFilmIzle/presence.ts:96]: let -> const

File: websites/I/IMDbFilmIzle/presence.ts [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+*/
+function getTimestamps(videoTime: number, videoDuration: number): Array<number> {
+  var startTime = Date.now();
+  var endTime = Math.floor(startTime / 1000) - videoTime + videoDuration;
+  return [Math.floor(startTime / 1000), endTime];
+}
+
+const video: { [k: string]: any } = {};
+
+presence.on("iFrameData", (data:any) => {
```
Comment (ririxichan) [inline websites/I/IMDbFilmIzle/presence.ts:40]: Change any to another type

File: websites/I/IMDbFilmIzle/presence.ts [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+  if (!data.error) {
+    video.dataAvailable = true;
+    video.currentTime = data.currentTime;
+    video.duration = data.duration;
+    video.paused = data.paused;
+  }
+});
+
+
```
Comment (ririxichan) [inline websites/I/IMDbFilmIzle/presence.ts:49]: delete this blank line.

**Review by ririxichan — CHANGES_REQUESTED** [review]

**Review by ririxichan — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Review by Alanexei — APPROVED** [review]

**Pull request merged**